### PR TITLE
histogram2d fixes

### PR DESCRIPTION
**Pull request by etpinard**
@plotly/plotly_js This PR starts with a fix for https://github.com/plotly/plotly.js/issues/3909 then follow-up with two more `histogram2d*` bug fixes.

### CODE REVIEW

File: src/traces/histogram/calc.js [hunk 1]
```diff
@@ -232,8 +232,8 @@ function calcAllAutoBins(gd, trace, pa, mainData, _overlayEdgeCase) {
 
     // all but the first trace in this group has already been marked finished
     // clear this flag, so next time we run calc we will run autobin again
-    if(trace._autoBinFinished) {
-        delete trace._autoBinFinished;
+    if(trace['_' + mainData + 'autoBinFinished']) {
```
Comment (etpinard) [inline src/traces/histogram/calc.js]: Fixup from https://github.com/plotly/plotly.js/issues/3909

As now both `histogram` and `histogram2d*` traces go through `calcAllAutoBins`, we need to split the "has autoBin finished` logic into x and y values.

File: test/image/mocks/histogram2d_bingroup-coloraxis.json [hunk 2]
```diff
@@ -0,0 +1,756 @@
+{
+  "data": [
```
Comment (etpinard) [inline test/image/mocks/histogram2d_bingroup-coloraxis.json]: before: https://codepen.io/nicolaskruchten/pen/MdVdGw
after: https://codepen.io/etpinard/pen/mYzbKE

File: test/image/mocks/histogram2dcontour_bingroup-coloraxis.json [hunk 3]
```diff
@@ -0,0 +1,756 @@
+{
+  "data": [
```
Comment (etpinard) [inline test/image/mocks/histogram2dcontour_bingroup-coloraxis.json]: This one looks like:

image

on master.

**Comment by archmoj** [comment]
Nicely done.
:dancer: :dancer:

**Pull request merged**